### ISSUE TITLE
Custom war packager Library breaks

### ISSUE THREAD

**Issue by sladyn98** [jenkinsci/custom-war-packager#156]
I am trying to use the jenkins war packager library in the custom distribution service. You can find the pull request here but when I run the build it errors out
Looking at this should provide more information
https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fcustom-distribution-service/detail/PR-41/13/pipeline/47#log-689

**Comment by oleg-nenashev** [comment]
```
[2020-06-16T15:27:59.580Z] [INFO] Running com.org.jenkins.custom.jenkins.distribution.service.generators.WarGeneratorTest

[2020-06-16T15:28:01.491Z] [WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/jenkins/workspace/ustom-distribution-service_PR-41/target/surefire-reports/2020-06-16T15-27-56_564-jvmRun1.dumpstream

[2020-06-16T15:28:01.768Z] java.io.IOException: Command failed with exit code 1: mvn clean package

[2020-06-16T15:28:01.768Z] 	at io.jenkins.tools.warpackager.lib.util.SystemCommandHelper.processFor(SystemCommandHelper.java:39)

[2020-06-16T15:28:01.768Z] 	at io.jenkins.tools.warpackager.lib.util.MavenHelper.run(MavenHelper.java:66)

[2020-06-16T15:28:01.768Z] 	at io.jenkins.tools.warpackager.lib.util.MavenHelper.run(MavenHelper.java:46)

[2020-06-16T15:28:01.769Z] 	at io.jenkins.tools.warpackager.lib.impl.Builder.build(Builder.java:146)

[2020-06-16T15:28:01.769Z] 	at com.org.jenkins.custom.jenkins.distribution.service.generators.WarGenerator.generateWAR(WarGenerator.java:28)
```

@sladyn98 From what I see you do not get a full Custom WAR Packager CLI output in your test. How  do you capture System logs and forked process streams there?

**Comment by sladyn98** [comment]
I am not too sure I am just using the library to generate. Here is the function to be precise
```
 public static void generateWAR(String versionName) {
        final Config cfg;
        Path tempDirWithPrefix = null;
        try {
            tempDirWithPrefix = Files.createTempDirectory(TEMP_PREFIX);
            final File configPath = util.getFileFromResources(PACKAGER_CONFIG_YAML);
            cfg = Config.loadConfig(configPath);
            cfg.buildSettings.setTmpDir(tempDirWithPrefix.toFile());
            cfg.buildSettings.setVersion(versionName);
            cfg.buildSettings.setInstallArtifacts(true);
            new Builder(cfg).build();
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            util.cleanupTempDirectory(tempDirWithPrefix.toFile());
        }
    }
```

**Comment by sladyn98** [comment]
@oleg-nenashev 
Steps to Reproduce 
```
git clone https://github.com/jenkinsci/custom-distribution-service.git
```

```
cd custom-distribution-service
```

```
mvn -Dtest=WarGeneratorTest test

```

**Comment by oleg-nenashev** [comment]
Ok, I will take a look

**Comment by oleg-nenashev** [comment]
@sladyn98 Your build does not use the Standard Jenkins parent POM. Instead of that it uses the Spring one. I will start from basic diagnostics since I was able to reproduce it locally. The Maven Surefire crash log includes the following message:

```
# Created at 2020-06-23T10:19:34.387
Corrupted STDOUT by directly writing to native stream in forked JVM 1. Stream '[INFO] --------------------------------[ jar ]---------------------------------'.
java.lang.IllegalArgumentException: Stream stdin corrupted. Expected comma after third character in command '[INFO] --------------------------------[ jar ]---------------------------------'.
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient$OperationalData.<init>(ForkClient.java:507)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.processLine(ForkClient.java:210)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.consumeLine(ForkClient.java:177)
	at org.apache.maven.plugin.surefire.booterclient.output.ThreadedStreamConsumer$Pumper.run(ThreadedStreamConsumer.java:88)
	at java.lang.Thread.run(Thread.java:748)
```

Analysis:

* Your build declares `maven-surefire-plugin:2.22.2`
* The issue is reported as https://issues.apache.org/jira/browse/SUREFIRE-1629 . You can easily find it by a stacktrace
* https://github.com/apache/maven-surefire/commit/c2a32917a82189a77750932aef821ac6a4cb2906 was merged into 3.0.0-M5, but AFAICT is was not backported into the 2.x branch
* Updating to Maven Surefire 3.x looks to be a reasonable solution. The Jenkins project already uses this version in its parent POMs.

I am closing the issue, because there is no evidence the issue is related to the Custom WAR Packager library

**Comment by sladyn98** [comment]
Thanks @oleg-nenashev
